mkosi/distributions/serpentos.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import re
+import shutil
+import textwrap
+from collections.abc import Sequence
+
+from mkosi.config import Architecture
+from mkosi.context import Context
+from mkosi.distributions import Distribution, DistributionInstaller, PackageType
+from mkosi.installer.moss import invoke_moss
+from mkosi.log import die
+
+
+class Installer(DistributionInstaller):
+    @classmethod
+    def pretty_name(cls) -> str:
+        return "Serpent OS"
+
+    @classmethod
+    def filesystem(cls) -> str:
+        return "btrfs"
+
+    @classmethod
+    def package_type(cls) -> PackageType:
+        return PackageType.moss
+
+    @classmethod
+    def default_release(cls) -> str:
+        return "rolling"
+
+    @classmethod
+    def default_tools_tree_distribution(cls) -> Distribution:
+        return Distribution.serpentos
+
+    @classmethod
+    def _move_kernel_and_initrd(cls, context: Context) -> None:
```

Comment:
I don't think we really need regexp for this. Please just do this similarly to how we do it for Debian:

```
for d in context.root.glob("boot/vmlinuz-*"):
    kver = d.name.removeprefix("vmlinuz-")
    vmlinuz = context.root / "usr/lib/modules" / kver / "vmlinuz"
    if not vmlinuz.exists():
        shutil.copy2(d, vmlinuz)
```